aws/resource_aws_backup_plan.go

```diff
@@ -36,10 +40,18 @@ func resourceAwsBackupPlan() *schema.Resource {
 						"rule_name": {
 							Type:     schema.TypeString,
 							Required: true,
+							ValidateFunc: validation.All(
+								validation.StringLenBetween(1, 50),
+								validation.StringDoesNotMatch(regexp.MustCompile(`^[a-zA-Z0-9\-\_\.]$`), ""),
```

Comment:
Shouldn't this validation be checking for all characters to match? This is currently checking that the string does not match a single character of these types.

Also its generally good to provide practitioners with a helpful validation error message rather than the raw regular expression. See also some future linting: https://github.com/bflad/tfproviderlint/issues/214

```suggestion
								validation.StringMatch(regexp.MustCompile(`^[a-zA-Z0-9\-\_\.]+$`), "must contain only alphanumeric characters, hyphens, underscores, and periods"),
```